src/app/content/practiceQuestions/hooks/locationChange.ts

```diff
@@ -38,19 +38,28 @@ const setSection = (services: MiddlewareAPI & AppServices, book: Book, page: Pag
   }
 };
 
-export const hookBody = (services: MiddlewareAPI & AppServices) => async() => {
+export const loadPracticeQuestionsSummaryHookBody = (services: MiddlewareAPI & AppServices) => async() => {
   const state = services.getState();
   const { book, page } = bookAndPage(state);
   const isEnabled = practiceQuestionsEnabled(state);
+  const summary = practiceQuestionsSummarySelector(state);
 
-  if (!book || !page || !isEnabled ) { return; }
+  if (!book || !page || !isEnabled || summary !== null) { return; }
 
-  if (practiceQuestionsSummarySelector(state) === null) {
-    const practiceQuestionsSummary = await loadSummary(services, book);
-    if (practiceQuestionsSummary) {
-      services.dispatch(receivePracticeQuestionsSummary(practiceQuestionsSummary));
-    }
+  const practiceQuestionsSummary = await loadSummary(services, book);
+  if (practiceQuestionsSummary) {
+    services.dispatch(receivePracticeQuestionsSummary(practiceQuestionsSummary));
   }
+};
+
+const hookBody = (services: MiddlewareAPI & AppServices) => async() => {
+  const state = services.getState();
+  const { book, page } = bookAndPage(state);
+  const isEnabled = practiceQuestionsEnabled(state);
+
+  if (!book || !page || !isEnabled) { return; }
+
+  await loadPracticeQuestionsSummaryHookBody(services)();
 
   setSection(services, book, page);
```

Comment:
won't this make it so that if location change fires before feature flags are received and then the dialog is opened the section won't be set?